src/panels/logbook/ha-panel-logbook.ts

```diff
@@ -197,13 +198,38 @@ export class HaPanelLogbook extends LitElement {
     }
   }
 
-  private async _fetchUsers() {
-    const users = await fetchUsers(this.hass);
-    const userid_to_name = {};
-    users.forEach((user) => {
-      userid_to_name[user.id] = user.name;
-    });
-    this._userIdToName = userid_to_name;
+  private async _fetchUserNames() {
+    const userIdToName = {};
+
+    // Start loading all the data
+    const personProm = fetchPersons(this.hass);
+    const userProm = this.hass.user!.is_admin && fetchUsers(this.hass);
+
+    // Process persons
+    const persons = await personProm;
+
+    for (const person of persons.storage) {
```

Comment:
```suggestion
    for (const person of [...persons.storage, ...persons.config]) {
```

Reply:
With that we're making slower code (copying things) to make code cleaner